pkg/gce-csi-driver/controller.go

```diff
@@ -0,0 +1,427 @@
+/*
+Copyright 2018 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+    http://www.apache.org/licenses/LICENSE-2.0
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package gceGCEDriver
+
+import (
+	"fmt"
+	"strings"
+
+	csi "github.com/container-storage-interface/spec/lib/go/csi/v0"
+	"github.com/golang/glog"
+	gce "github.com/kubernetes-sigs/gcp-compute-persistent-disk-csi-driver/pkg/gce-cloud-provider"
+	utils "github.com/kubernetes-sigs/gcp-compute-persistent-disk-csi-driver/pkg/utils"
+	"golang.org/x/net/context"
+	compute "google.golang.org/api/compute/v1"
+	"google.golang.org/grpc/codes"
+	"google.golang.org/grpc/status"
+)
+
+// TODO: Add noisy glog.V(5).Infof() EVERYWHERE
+// TODO: Improve errors to only expose codes at top level
+// TODO: Improve error prefix to explicitly state what function it is in.
+
+type GCEControllerServer struct {
+	Driver        *GCEDriver
+	CloudProvider gce.GCECompute
+}
+
+const (
+	// MaxVolumeSize is the maximum standard and ssd size of 64TB
+	MaxVolumeSize       int64 = 64000000000000
+	DefaultVolumeSize   int64 = 5000000000
+	MinimumDiskSizeInGb       = 5
+
+	DiskTypeSSD      = "pd-ssd"
+	DiskTypeStandard = "pd-standard"
+	diskTypeDefault  = DiskTypeStandard
+
+	diskTypePersistent = "PERSISTENT"
```

Comment:
Seems like there's two kinds of "disk types".  Can you rename one of them to be disambiguate them?